### PR TITLE
GumGum adapter

### PR DESCRIPTION
**Pull request by davidmh**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: 'gumgum',
  params: {
    inScreen: 'tracking_id'
  }
  // More products will be incorporated in the future.
}
```

- contact email of the adapter’s maintainer
dmejorado@gumgum.com
- [x] official adapter submission

### CODE REVIEW

**Comment by kwunchoy** [comment]
@matthewlane sorry to bother you but can help us get our PR reviewed?

**Comment by matthewlane** [comment]
Yes, I'll review this today. Thanks for your patience

File: src/adapters/gumgum.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      const { t: trackingId
+            , pi: productId
+            , placementCode
+            } = cachedBidRequest;
+      bidResponse.placementCode = placementCode;
+      const encodedResponse = encodeURIComponent(JSON.stringify(bidResponse));
+      const gumgumAdLoader = `r("${trackingId}",${productId},"${ encodedResponse }",window,top)`;
+      const gumgumLibrary = `(function(r){${gumgumAdLoader}})(function(trackingId,prodId,data,context,w,d,s,G){` +
+        'G=w.GUMGUM;d=w.document;function lg(){w.GUMGUM.pbjs(trackingId,prodId,data,context,this)}return!G?w' +
+        '.$$PREBID_GLOBAL$$.loadScript("https://g2.gumgum.com/javascripts/ggv2.js",lg):lg()});';
```
Comment (matthewlane) [inline src/adapters/gumgum.js]: Line 77 - 79 is hard for me to understand, can you clarify what's happening here? The rest of your code style looks great, if it's possible to apply that to this section and retain functionality, please do

File: test/spec/adapters/gumgum_spec.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  function mockBidResponse(response) {
+    sandbox.stub(bidManager, 'addBidResponse');
+    sandbox.stub(adLoader, 'loadScript');
+    adapter.callBids(bidderRequest);
+    pbjs.handleGumGumCB['InScreenBidId'](response);
+    return bidManager.addBidResponse.firstCall.args[1];
+  }
+
+  beforeEach(() => {
+    pbjs._bidsRequested = [];
```
Comment (davidmh) [inline test/spec/adapters/gumgum_spec.js]: I don't think we need it. I was just trying to follow the convention used on other tests.

I'll remove it shortly.

File: src/adapters/gumgum.js (hunk 1 above)
Comment (davidmh) [inline src/adapters/gumgum.js]: A lot of our creatives assume the GumGum script is running on `top`, so this function is intended to pass our winning bid, to the entry point in our script (`top.GUMGUGM.pbjs`), but first we need to be sure our script exists on `top`.

Here's the uncompressed script:

```javascript
(function (request) {
    request( /* tracking id */ , /* productId */ , /* ad payload */ , /* iframed window */ , /* top window*/ );
})(function (trackingId, prodId, data, context, w, d, s, G) {
    G = w.GUMGUM;
    d = w.document;
    // load gumgum script
    function lg() {
        w.GUMGUM.pbjs(trackingId, prodId, data, context, this);
    }
    return !G
        // if the gumgum object doesn't exist on the top window, we need to load it
        // and then pass the payload
        ? w.$$PREBID_GLOBAL$$.loadScript("https://g2.gumgum.com/javascripts/ggv2.js", lg)
        // if the script is already running, just pass the payload
        : lg();
});
```

Comment (davidmh) [inline src/adapters/gumgum.js]: @matthewlane do you need me to change the format?

Comment (mkendall07) [inline src/adapters/gumgum.js]: @davidmh 
Please include the unminified version in the adapter code. Thanks

Comment (davidmh) [inline src/adapters/gumgum.js]: Done

**Pull request merged**